packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -1443,6 +1414,113 @@ actual object RealmInterop {
     }
 }
 
+/**
+ * A factory and container for various resources that can be freed when calling [free].
+ *
+ * @see memScope
+ */
+private class MemScope {
+    val values: MutableSet<realm_value_t> = mutableSetOf()
+
+    fun managedRealmValue(value: RealmValue): realm_value_t {
+        val element = capiRealmValue(value)
+        values.add(element)
+        return element
+    }
+
+    fun free() {
+        values.map { realmc.realm_value_t_cleanup(it) }
+    }
+}
+
+/**
+ * A scope providing a [MemScope] that collects the created resources and automatically frees them
+ * upon exiting the scope.
+ */
+private fun <R> memScope(block: MemScope.() -> R): R {
+    val scope = MemScope()
+    val result: R = block(scope)
+    scope.free()
+    return result
+}
+
+// TODO OPTIMIZE Maybe move this to JNI to avoid multiple round trips for allocating and
+//  updating before actually calling
+@Suppress("ComplexMethod", "LongMethod")
+private fun capiRealmValue(realmValue: RealmValue): realm_value_t {
+    val cvalue = realm_value_t()
+    val value = realmValue.value
+    if (value == null) {
+        cvalue.type = realm_value_type_e.RLM_TYPE_NULL
+    } else {
+        when (value) {
+            is String -> {
+                cvalue.type = realm_value_type_e.RLM_TYPE_STRING
+                cvalue.string = value
+            }
+            /*is Byte -> {
```

Comment:
Why this is still commented? there's no use case for these managed types yet?

Reply:
I just moved the `to_realm_value` method around, so didn't really update anything. We are not passing other types than the `RealmStorageValue` types in/out after the #790. Will just remove them.